android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt

```diff
@@ -223,113 +220,126 @@ class MusicService : HeadlessJsTaskService() {
         return compactCapabilities.contains(capability)
     }
 
+    @MainThread
     fun add(track: Track) {
         add(listOf(track))
     }
 
+    @MainThread
     fun add(tracks: List<Track>) {
         val items = tracks.map { it.toAudioItem() }
-        scope.launch { player.add(items, false) }
+        player.add(items, false)
     }
 
+    @MainThread
     fun add(tracks: List<Track>, atIndex: Int) {
         val items = tracks.map { it.toAudioItem() }
-        scope.launch { player.add(items, atIndex) }
+        player.add(items, atIndex)
     }
 
+    @MainThread
     fun remove(index: Int) {
         remove(listOf(index))
     }
 
-    fun remove(indexes: List<Int>) = scope.launch {
+    @MainThread
+    fun remove(indexes: List<Int>) {
         player.remove(indexes)
     }
 
-    fun play() = scope.launch {
+    @MainThread
+    fun play() {
         player.play()
     }
 
-    fun pause() = scope.launch {
+    @MainThread
+    fun pause() {
         player.pause()
     }
 
-    fun stop() = scope.launch {
+    @MainThread
+    fun stop() {
         player.stop()
     }
 
-    fun removeUpcomingTracks() = scope.launch {
+    @MainThread
+    fun removeUpcomingTracks() {
         player.removeUpcomingItems()
     }
 
-    fun removePreviousTracks() = scope.launch {
+    @MainThread
+    fun removePreviousTracks() {
         player.removePreviousItems()
     }
 
-    fun skip(index: Int) = scope.launch {
+    @MainThread
+    fun skip(index: Int) {
         player.jumpToItem(index, player.isPlaying)
     }
 
-    fun skipToNext() = scope.launch {
+    @MainThread
+    fun skipToNext() {
         player.next()
     }
 
-    fun skipToPrevious() = scope.launch {
+    @MainThread
+    fun skipToPrevious() {
         player.previous()
     }
 
-    fun seekTo(seconds: Float) = scope.launch {
+    @MainThread
+    fun seekTo(seconds: Float) {
         player.seek((seconds * 1000).toLong(), TimeUnit.MILLISECONDS)
     }
 
-    fun getCurrentTrackIndex(callback: (Int) -> Unit) = scope.launch {
-        callback(player.currentIndex)
-    }
+    @MainThread
+    fun getCurrentTrackIndex(): Int = player.currentIndex
 
-    fun getRate(callback: (Float) -> Unit) = scope.launch {
-        callback(player.playbackSpeed)
-    }
+    @MainThread
```

Comment:
I still think we should remove these here, since services always run on main thread 
cc @dcvz

Reply:
@dcvz this is the only thing hanging up this PR. Would you be able to take a look?

Reply:
@mpivchev technically someone modifying the codebase could call these service methods from any thread they wanted to when extending the code. I think these annotations are a good safety in the event that that happens.

My 2 cents is to merge this and be done with it.